### PR TITLE
[Merged by Bors] - Add right to create pods/eviction objects to the clusterrole to allow the restart operator to restart pods.

### LINKED ISSUE #66: The restart controller is not allowed to evict pods on GKE (and potentially other k8s flavors) due to missing rights in the clusterrole
### Affected version

0.2.0

### Current and expected behavior

Expected:
When a pod reaches the expiration date specified by the annotation '' it should be restarted by the restart controller.

Current:
The pod is not restarted, and the restart controller logs the following error in its log:

```
Failed to reconcile object controller.name="pod.restarter.commons.stackable.tech" error=reconciler for object Pod.v1./simple-nifi-node-default-0.default failed error.sources=[failed to evict Pod, ApiError: pods "simple-nifi-node-default-0" is forbidden: User "system:serviceaccount:default:commons-operator-serviceaccount" cannot create resource  │
│ "pods/eviction" in API group "" in the namespace "default": Forbidden (ErrorResponse { status: "Failure", message: "pods \"simple-nifi-node-default-0\" is forbidden: User \"system:serviceaccount:default:commons-operator-serviceaccount\" cannot create resource \"pods/eviction\" in API group \"\" in the namespace \"default\"", reason: "Forbidden", code: 403 }), pods "simple-nifi-node-default-0" is forbidden: User "syste │
│ m:serviceaccount:default:commons-operator-serviceaccount" cannot create resource "pods/eviction" in API group "" in the namespace "default": Forbidden]
```

### Possible solution

The restart controller needs to be granted `create` rights on the 'pods/eviction' objects.
To achieve this the following should be added to the clusterrole for this deployment:

```
  - apiGroups:
      - ""
    resources:
      - pods/eviction
    verbs:
      - create
```

### Additional context

_No response_

### Environment

_No response_

### Would you like to work on fixing this bug?

yes

### PR DESCRIPTION
**Pull request by soenkeliebau**
# Description

fixes #66 

## Review Checklist

- [x] Code contains useful comments
- [x] CRD change approved (or not applicable)
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
- [x] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

**Review by sbernauer — APPROVED** [review]
LGTM % checks

**Comment by sbernauer** [comment]
Can you please add a Changelog entry?

File: CHANGELOG.md
Comment (sbernauer) [inline CHANGELOG.md:19]:
```suggestion
[#67]: https://github.com/stackabletech/commons-operator/pull/67
```

**Review by sbernauer — APPROVED** [review]

**Comment by sbernauer** [comment]
bors cancel

**Comment by soenkeliebau** [comment]
> Did you just ignore my discussion? :P (small typo in change-log)

Oh, sorry. I saw your comment, but thought I'd fixed it already. I totally missed that you were commenting on my updated commit already.

**Comment by sbernauer** [comment]
I could have also requested change instead of approval but i didn't want to be to picky and slow things down. So partly my fault as well ;)

**Comment by soenkeliebau** [comment]
I think we do not have this repo set up so that new commits dismiss exiisting approvals..

**Comment by sbernauer** [comment]
ahh didn't know that's a setting. I thought it's just random-ish

**Comment by soenkeliebau** [comment]
It sort of is I guess, just random over who set up the protection rules :)

So ..... is this ready for merging now?

**Comment by sbernauer** [comment]
> "message":"At least 1 approving review is required by reviewers with write access."

uhh

**Review by sbernauer — APPROVED** [review]
Ready for merging when you are

**Review by sbernauer — APPROVED** [review]
Shut up and take my approval

**Pull request closed without merging**
